src/main/java/org/example/homework5/NumberGenerator.java

```diff
@@ -0,0 +1,41 @@
+package org.example.homework5;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class NumberGenerator {
+    public static void main(String[] args) {
+        List<Integer> randomArray = generateRandomArray();
+        arraySum(randomArray);
+    }
+    public static List<Integer> generateRandomArray() {
+        List<Integer> array = new ArrayList<Integer>();
+        boolean flag = true;
+
+        while (flag) {
+            Random random = new Random();
+            int randomNumber = random.nextInt(100) + 1;
+
+            if (array.indexOf(randomNumber) >= 0) {
+                flag = false;
+            } else {
+                array.add(randomNumber);
+            }
+        }
+
+        System.out.println("Array of random generated numbers: " + array);
+
+        return array;
+    }
+
+    public static void arraySum(List<Integer> randomArray) {
```

Comment:
Actually using array here is also not required you can calculate the sum inside the loop

Reply:
Fixed